src/asynk/async_queue.rs

```diff
@@ -552,19 +482,31 @@ where
         transaction: &mut Transaction<'_>,
         metadata: serde_json::Value,
         task_type: &str,
+        scheduled_at: DateTime<Utc>,
     ) -> Result<Task, AsyncQueueError> {
-        match Self::find_task_by_metadata_query(transaction, &metadata).await {
+        match Self::find_task_by_uniq_hash_query(transaction, &metadata).await {
             Some(task) => Ok(task),
-            None => Self::insert_task_query(transaction, metadata, task_type).await,
+            None => {
+                Self::insert_task_uniq_query(transaction, metadata, task_type, scheduled_at).await
+            }
         }
     }
 
-    async fn find_task_by_metadata_query(
+    fn calculate_hash(json: String) -> String {
```

Comment:
I think you can just &str as a parameter

Reply:
[Comments adressed](https://github.com/ayrat555/fang/pull/71/commits/e32b0fe5b77adb1ff9ec8a9453335b37a00bfccc)